### DISCUSSION TITLE
Integral behaviour during overshot

### DISCUSSION THREAD

**Discussion by beltracri** [ScratMan/HASmartThermostat#246] [Q&A]
Hi there, I'm trying to set up a very slow system for underfloor water heating. I actually run on **kp : 180, ki : 0.001 and kd : 40000** and experiencing a strange behaviour related to integral component during the overshot.
It keeps addin up every cycle and, during the overshot, it decreases only of one small step (when first sampling occours with negative error).
I don't think this is a correct behaviour, i think that the integral has to decrease during all the overshot time!
With this behaviour, my integral keep increase day by day...

**Comment by maxeitelwein** [comment]
The issue is, that your system has a big dead time and the PID controller can not really cope with that. I would try to not use the I and D part and just a P part. Check out https://github.com/ScratMan/HASmartThermostat/discussions/260 and try to simulate your room temperature.
